wp-modules/app/js/src/hooks/useVersionControl.ts

```diff
@@ -0,0 +1,52 @@
+import { useState } from '@wordpress/element';
+import { patternManager } from '../globals';
+import getHeaders from '../utils/getHeaders';
+import type { Theme } from '../types';
+
+export default function useVersionControl(
+	initialDismissedThemes: Array< Theme[ 'name' ] >
+) {
+	const [ dismissedThemes, setDismissedThemes ] = useState<
+		Array< Theme[ 'name' ] >
+	>( initialDismissedThemes );
+
+	const warningShouldShow =
+		! Boolean( patternManager.versionControl ) &&
+		! dismissedThemes.some(
+			( dismissedTheme ) => dismissedTheme === patternManager.themeName
+		);
+
+	function updateDismissedThemes( themeNameToDismiss: Theme[ 'name' ] ) {
+		return fetch(
+			patternManager.apiEndpoints.updateDismissedThemesEndpoint,
+			{
+				method: 'POST',
+				headers: getHeaders(),
+				body: JSON.stringify( {
+					themeNames: dismissedThemes.reduce(
```

Comment:
This shouldn't need to pass anything in the `body`.

The back-end can get the current theme, and add that to the user meta array of dismissed themes.

Reply:
Though it's weird if POST request has no body.

Reply:
An alternative would be to pass only the active theme name in the `body`. But that might be needless.

Reply:
This logic has been greatly simplified in [37afb1a](https://github.com/studiopress/pattern-manager/pull/145/commits/37afb1a1d1a8d1d7419462840ca3fdb263ef7b38).

I like passing `themeName` in the body, but we can remove it if preferable.

Reply:
Just realized that it is not possible to pass `themeName` since we are no longer hydrating the app with that string.

Going to roll with no body for the time being.

Reply:
Body removed in [54f8656](https://github.com/studiopress/pattern-manager/pull/145/commits/54f8656b0520c814014a8f959ca53a160855de5c).

The `Notice` `handleDismiss` callback looks much cleaner now:

<img width="470" alt="Screenshot 2023-04-27 at 4 31 26 PM" src="https://user-images.githubusercontent.com/108079556/234995784-12ca988e-6dd8-4a21-aaa2-b6d08504e9b4.png">